graphblas_algorithms/nxapi/core.py

```diff
@@ -0,0 +1,13 @@
+from graphblas_algorithms import algorithms
+from graphblas_algorithms.classes.graph import to_undirected_graph
+from graphblas_algorithms.utils import not_implemented_for
+
+__all__ = ["k_truss"]
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+def k_truss(G, k):
+    G = to_undirected_graph(G, dtype=bool)
+    result = algorithms.k_truss(G, k)
+    return result.to_networkx()
```

Comment:
A comment for future us: I think it would be best if functions in `nxapi` namespace could still return e.g. a `graphblas_algorithms.Graph` instead of converting to a networkx-like object.  I think this also means creating a `NodeMap` that uses a `graphblas.Vector` and is a `abc.Mapping`, and, similarly, a `NodeSet` that is a `abc.Set`.  This should allow us to pass networkx tests.

The current PR is a refactor, so creating new objects like `NodeMap` is out of scope.